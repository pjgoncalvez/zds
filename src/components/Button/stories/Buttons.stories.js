import React from "react";
import { Button } from "../index";
import { colours, Theme } from "../../../components/theme";

export default {
  title: "Buttons",
  component: Button,
  argTypes: {
    text: { name: "Text", required: true },
    selectedTheme: {
      name: "Theme",
      required: false,
      defaultValue: "light",
      control: {
        type: "select",
        options: ["light", "dark", ""],
      },
    },
    size: {
      name: "Size",
      required: false,
      defaultValue: "giant",
      control: {
        type: "select",
        options: ["giant", "large", "medium", "small", "tiny"],
      },
    },
  },
  parameters: {
    backgrounds: {
      default: "light",
      values: [
        { name: "light", value: colours.light },
        { name: "dark", value: colours.dark },
      ],
    },
  },
};

const Template = (args) => (
  <Theme>
    <Button {...args}></Button>
  </Theme>
);

export const SimpleButton = Template.bind({});
SimpleButton.args = {
  text: "Just a simple button",
  size: "giant",
  selectedTheme: "light",
};

