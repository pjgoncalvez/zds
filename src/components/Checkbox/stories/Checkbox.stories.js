import React from "react";
import Checkbox from "../Checkbox";
import { colours } from "../../theme/";

export default {
  title: "Checkbox",
  component: Checkbox,
  argTypes: {
    checked: {
      name: "Checked",
      required: false,
      defaultValue: true,
      control: {
        type: "boolean",
        options: [true, false],
      },
    },
  },
  parameters: {
    backgrounds: {
      default: "white",
      values: [
        { name: "light", value: colours.light },
        { name: "dark", value: colours.dark },
        { name: "white", value: "#FFFFFF" },
      ],
    },
  },
};

const CheckboxStoryTemplate = (args) => <Checkbox {...args}></Checkbox>;

export const CheckboxStory = CheckboxStoryTemplate.bind();
CheckboxStory.args = {
  checked: false,
};
