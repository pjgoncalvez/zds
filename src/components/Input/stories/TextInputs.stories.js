import React from "react";
import { Input } from "../index";
import { colours } from "../../../components/theme";

export default {
  title: "Inputs",
  component: Input,
  argTypes: {
    disabled: {
      name: "Disabled",
      required: false,
      defaultValue: false,
      control: {
        type: "boolean",
        options: [true, false],
      },
    },
    value: {
      name: "Text",
      required: false,
      control: {
        type: "text",
      },
    },
    primaryLabel: {
      defaultValue: false,
      name: "Primary label",
      control: {
        type: "text",
      },
    },
    secondaryLabel: {
      defaultValue: false,
      name: "Secondary label",
      control: {
        type: "text",
      },
    },
    caption: {
      defaultValue: false,
      name: "Caption",
      control: {
        type: "text",
      },
    },
    placeholder: {
      name: "Placeholder",
      required: false,
      defaultValue: "Placeholder text",
      control: {
        type: "text",
      },
    },
    status: {
      name: "Status",
      required: false,
      control: {
        type: "select",
        options: ["success", "info", "warning", "error", "-"],
      },
    },
  },
  parameters: {
    backgrounds: {
      default: "white",
      values: [
        { name: "light", value: colours.light },
        { name: "dark", value: colours.dark },
        { name: "white", value: "#FFFFFF" },
      ],
    },
  },
  type: "text",
};

const SimpleTextInputTemplate = (args) => <Input {...args}></Input>;

export const SimpleTextInput = SimpleTextInputTemplate.bind();
SimpleTextInput.args = {
  value: "Input Text",
  disabled: false,
  placeholder: "Placeholder text",
  status: "",
};
