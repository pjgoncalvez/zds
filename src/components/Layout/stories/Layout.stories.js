import React from "react";
import { Row, Column, Grid } from "../../Layout";
import { colours, Theme } from "../../theme";

export default {
  title: "Basic Column",
  component: Column,
  argTypes: {
    width: {
      description: "Expects a integer value [1-12]",
      type: {
        name: "number",
        required: false,
      },
      defaultValue: 12,
      control: {
        type: "range",
        min: 1,
        max: 12,
        step: 1,
      },
    },
    text: { name: "text", required: false },
    selectedTheme: {
      name: "Theme",
      required: false,
      defaultValue: "light",
      control: {
        type: "select",
        options: ["light", "dark", ""],
      },
    },
    backgroundColor: {
      name: "Custom background color",
      control: "color",
    },
    children: {
      table: {
        disable: true,
      },
    },
  },
  parameters: {
    backgrounds: {
      default: "light",
      values: [
        { name: "light", value: colours.light },
        { name: "dark", value: colours.dark },
      ],
    },
  },
};

const Template = (args) => (
  <Theme>
    <Grid>
      <Row>
        <Column {...args}>
          <h2>hello</h2>
        </Column>
      </Row>
    </Grid>
  </Theme>
);

export const SingleColumn = Template.bind({});
SingleColumn.args = {
  width: 12,
  text: "Vivamus nizzle maurizzle my shizz tellivizzle i'm in the shizzle",
  backgroundColor: "#F6F5DF",
  selectedTheme: "light",
};
