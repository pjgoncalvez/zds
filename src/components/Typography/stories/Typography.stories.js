import React from "react";
import { H1, H2, H3, H4, H5, H6 } from "../Typography";
import { colours } from "../../../components/theme";

export default {
  title: "Typography",
  component: H1,
  argTypes: {},
  parameters: {
    backgrounds: {
      default: "white",
      values: [
        { name: "light", value: colours.light },
        { name: "dark", value: colours.dark },
        { name: "white", value: "#FFFFFF" },
      ],
    },
  },
};

const Heading1Template = (args) => (
    <H1 {...args} />
);

export const Heading1 = Heading1Template.bind();
Heading1.args = {
  text:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry.",
};

const Heading2Template = (args) => (
    <H2 {...args} />
);

export const Heading2 = Heading2Template.bind();
Heading2.args = {
  text:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry.",
};
