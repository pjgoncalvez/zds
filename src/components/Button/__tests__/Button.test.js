// Button.test.js

import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

import { SimpleButton } from "../stories/Buttons.stories";

it("renders a default button", () => {
  render(<SimpleButton {...SimpleButton.args} />);
  expect(screen.getByRole("button")).toHaveTextContent("Just a simple button");
});

it("matches the expected snapshot", () => {
  const tree = renderer
    .create(<SimpleButton {...SimpleButton.args} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
