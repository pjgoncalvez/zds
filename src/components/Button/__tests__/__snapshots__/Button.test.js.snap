// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the expected snapshot 1`] = `
.c0 {
  width: auto;
  cursor: pointer;
  text-align: center;
  line-height: 1em;
  height: 56px;
  font-size: 24px;
  padding-left: 26px 22px;
  padding-right: 26px 22px;
}

<button
  className="c0"
  size="giant"
>
  Just a simple button
</button>
`;
